pkg/configuration/base/resources/service.go

```diff
@@ -70,11 +70,9 @@ func GetJenkinsSlavesServiceFQDN(jenkins *v1alpha2.Jenkins) (string, error) {
 }
 
 // GetClusterDomain returns Kubernetes cluster domain, default to "cluster.local"
-func getClusterDomain() (string, error) {
-	clusterDomain := "cluster.local"
-
-	if ok, err := isRunningInCluster(); !ok {
-		return clusterDomain, nil
+func getClusterDomain(kubernetesClusterDomain string) (string, error) {
+	if isRunningInCluster, err := isRunningInCluster(); !isRunningInCluster {
+		return kubernetesClusterDomain, nil
 	} else if err != nil {
 		return "", nil
 	}
```

Comment:
don't use `else if`, if you return something in first condition

This code looks good, but I think it can be simpler
```
runningInCluster, err := isRunningInCluster();
if !runningInCluster {
	return kubernetesClusterDomain, nil
}
 
if err != nil {
	return "", nil
}
```

Reply:
it's up to you, which version you choose

Reply:
As you can see it's not my part of the code, but you're right. It needs a refactor.